src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,108 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private static final int DEFAULT_CAPACITY = 16;
+    private Node<K, V>[] array;
+    private int threshhold;
+    private int size;
+
+    public MyHashMap() {
+        array = (Node<K,V>[]) new Node[DEFAULT_CAPACITY];
+        threshhold = (int) (DEFAULT_CAPACITY * DEFAULT_LOAD_FACTOR);
+    }
+
+    @Override
+    public int getSize() {
+        return size;
+    }
+
+    private int hash(K key) {
+        int h;
+        return (key == null) ? 0 : (h = key.hashCode()) ^ (h >> 16);
+    }
 
     @Override
     public void put(K key, V value) {
+        int keyHash = hash(key);
+        int index = keyHash % array.length;
+        Node<K, V> node = new Node<>(keyHash, key, value, null);
+        put(array, node, index);
+        if (size > threshhold) {
+            resize();
+        }
+    }
+
+    private void put(Node<K, V>[] toArray, Node<K, V> newNode, int index) {
```

Comment:
```suggestion
    private void put(Node<K, V> newNode, int index) {
```
we don't need to pass arr as input, since we can access class field directly. If you need different arrays in the resize method - just create a temp one there to isolate this logic